dev/sg/sg_feedback.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/urfave/cli/v2"
+
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/open"
+	"github.com/sourcegraph/sourcegraph/dev/sg/internal/std"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"github.com/sourcegraph/sourcegraph/lib/output"
+)
+
+const newDiscussionURL = "https://github.com/sourcegraph/sourcegraph/discussions/new"
+
+func addFeedbackSubcommand(commands []*cli.Command) {
+	giveFeedback := false
+	feedbackFlag := cli.BoolFlag{
+		Name:        "feedback",
+		Usage:       "provide feedback about this command by opening up a Github discussion",
+		Destination: &giveFeedback,
+	}
+
+	for _, command := range commands {
+		command.Flags = append(command.Flags, &feedbackFlag)
+		action := command.Action
+		command.Action = func(ctx *cli.Context) error {
+			if giveFeedback {
+				return feedbackExec(ctx)
+			}
+			return action(ctx)
+		}
+
+		addFeedbackSubcommand(command.Subcommands)
+	}
+}
+
+var feedbackCommand = &cli.Command{
+	Name:     "feedback",
+	Usage:    "opens up a Github discussion page to provide feedback about sg",
+	Category: CategoryCompany,
+	Action:   feedbackExec,
+}
+
+func feedbackExec(ctx *cli.Context) error {
+	std.Out.WriteLine(output.Styledf(output.StylePending, "Gathering feedback for sg %s ...", ctx.Command.FullName()))
+	title, body, err := gatherFeedback(ctx, std.Out, os.Stdin)
+	if err != nil {
+		return err
+	}
+	body = addSGInformation(ctx, body)
+
+	if err := sendFeedback(ctx.Context, title, "developer-experience", body); err != nil {
+		return err
+	}
+	return nil
+}
+
+func gatherFeedback(ctx *cli.Context, out *std.Output, in io.Reader) (string, string, error) {
+	out.Promptf("Write your feedback below and press <CTRL+D> when you're done.\n")
+	body, err := io.ReadAll(in)
+	if err != nil && err != io.EOF {
+		return "", "", err
+	}
+
+	out.Promptf("The title of your feedback is going to be \"sg %s\". Anything else you want to add ?\n", ctx.Command.FullName())
+	reader := bufio.NewReader(in)
+	userTitle, err := reader.ReadString('\n')
+	if err != nil {
+		return "", "", err
+	}
+
+	title := "sg " + ctx.Command.FullName()
+	userTitle = strings.TrimSpace(userTitle)
+	switch strings.ToLower(userTitle) {
+	case "":
+	case "na":
+	case "no":
+	case "nothing":
+	case "nope":
```

Comment:
trivia - you can also inline this:

```suggestion
	case "", "na", "no", "nothing", "nope":
```

Mostly a stylistic choice though!